pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -471,6 +481,13 @@ func (s *Reconciler) Delete(ctx context.Context) error {
 		}
 	}
 
+	availabilitySetName := s.scope.Machine.Labels[MachineSetLabelName]
+	if err := s.availabilitySetsSvc.Delete(ctx, availabilitysets.Spec{
```

Comment:
what happens if there are multiple machines in the availability set? (or is that not possible)

Reply:
Multiple machines is definitely possible and in fact intended, do we not need to check first that the AS is empty before removing it?

Reply:
The deletion logic for availability set is almost a plain copy-paste from upstream CAPZ